python/pyspark/sql/tests.py

```diff
@@ -2461,17 +2461,13 @@ def test_join_without_on(self):
         df1 = self.spark.range(1).toDF("a")
         df2 = self.spark.range(1).toDF("b")
 
-        try:
-            self.spark.conf.set("spark.sql.crossJoin.enabled", "false")
+        with self.sql_conf({"spark.sql.crossJoin.enabled": False}):
             self.assertRaises(AnalysisException, lambda: df1.join(df2, how="inner").collect())
 
-            self.spark.conf.set("spark.sql.crossJoin.enabled", "true")
+        with self.sql_conf({"spark.sql.crossJoin.enabled": True}):
```

Comment:
So the `sql_conf` context will change this back to be unset right?

Reply:
Yup, it originally unset `spark.sql.crossJoin.enabled` but now it set to the original value back.
If `spark.sql.crossJoin.enabled` is unset in this test, it will change this back to be unset.